src/os/data/idatadescriptor.js

```diff
@@ -268,6 +268,12 @@ os.data.IDataDescriptor.prototype.getSearchText;
 os.data.IDataDescriptor.prototype.getNodeUI;
 
 
+/**
+ * Sets the Node UI for this desceriptor
+ * @param {string} node UI HTML
+ */
+os.data.IDataDescriptor.prototype.setNodeUI;
```

Comment:
`setNodeUI` will need to be implemented anywhere that extends `os.data.IDataDescriptor` ... so `iogcdescriptor.js`, `istatedescriptor.js`, `iarcdescriptor.js` and anything downstream from there

Reply:
@welchyd So this will be a breaking change for external plugins?

Reply:
Yes ... @robertbolan-bits might be able to just limit it to `baseDescriptor.js`, but yes it would affect external plugins otherwise

Reply:
Depending on how that works out, https://opensphere.readthedocs.io/en/latest/guides/plugin_file_type_guide/descriptor.html (https://github.com/ngageoint/opensphere/blob/master/docs/guides/plugin_file_type_guide/descriptor.rst and included demo code) might need an update.

Reply:
I will move the new function to just the base descriptor and take it off the interface. I think that will be for the best so the change isnt so wide spread